src/dug/core/parsers/radx_parser.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from typing import List
+from xml.etree import ElementTree as ET
+
+from dug import utils as utils
+from ._base import DugElement, FileParser, Indexable, InputFile
+
+logger = logging.getLogger('dug')
+
+
+class RADxParser(FileParser):
+
+    def __call__(self, input_file: InputFile) -> List[Indexable]:
+        tree = ET.parse(input_file, ET.XMLParser(encoding='utf-8'))
+        root = tree.getroot()
+        study_id = root.attrib['id']
+        # Parse study name from GapExchange file, and if that fails try from file handle
+        # If still None, raise an error message
+        study_name = root.attrib['study_name']
+        elements = []
+        for variable in root.iter('variable'):
+            desc = variable.find('description').text if variable.find('description') is not None else ''

```

Comment:
I usually prefer `.findall()` instead of `.find()` because it finds all the elements instead of just the first one, and because then I can just loop over all the descriptions it found instead of testing for `if variable.find('description') is not None`. However, if you're pretty sure that there will only be a single description, then this should be fine!